File: Solution/1089.复写零/1089.复写零.py
'''
Descroption: LeetCode 1089. 复写零
Author: EmoryHuang
Date: 2022-06-17 20:38:30
Method:
简单题，遍历数组，如果是 0，则插入 1 并 pop
'''


class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        i = 0
        while i < len(arr):
            if arr[i] == 0:
                arr.insert(i, 0)
                arr.pop()
                i += 1
            i += 1

File: Solution/1089.复写零/1089.复写零.cpp
/*
 * @Descroption: LeetCode 1089. 复写零
 * @Author: EmoryHuang
 * @Date: 2022-06-17 20:41:58
 * @Method:
 * 简单题，遍历数组，如果是 0，则插入 1 并 pop
 */

class Solution {
   public:
    void duplicateZeros(vector<int>& arr) {
        int i = 0, n = arr.size();
        while (i < n) {
            if (arr[i] == 0) {
                arr.insert(arr.begin() + i, 0);
                arr.erase(arr.end() - 1);
                i++;
            }
            i++;
        }
    }
};